spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/NacosDiscoveryProperties.java

```diff
@@ -260,17 +275,43 @@ public void init() throws Exception {
 		namespace = Objects.toString(namespace, "");
 		logName = Objects.toString(logName, "");
 
-		if (StringUtils.isEmpty(ip)) {
-			// traversing network interfaces if didn't specify a interface
+		InetUtils.HostInfo hostInfo;
+
+		if (StringUtils.isBlank(host) && StringUtils.isNotBlank(ip)) {
```

Comment:
if !StringUtils.isEmpty(host)&&StringUtils.isNotBlank(ip), I think use ip is more reasonable. It can't cause confused to old users. If a user want to use host, just only when he/she set host without ip.